### ISSUE TITLE
How Manu client for server ?

### ISSUE THREAD

**Issue by zoic21** [mmatczuk/go-http-tunnel#113]
Hello

I wonder how many client can be handle by server ? I work on project with about 10000 client. I imagine use multi server container server with ha proxy to redirect DNS on good server. Someone try this ?

Thank in advance for response.

**Comment by faelp22** [comment]
Hello everyone, I would also like to know how many customers does tuneld support simultaneously?

**Comment by mperktold** [comment]
When trying to connect two clients to a single server, the server rejects the second client.
So apparently, only one client is supported.

Would it be possible to add support for multiple client?

I am looking for a solution where a public server acts as both a tunnel and a dispatcher for multiple servers that are behind a firewall. The dispatching could be done using virtual hostnames, or based on the authentification, but given a corresponding hook and basic support, I could implement the dispatch logic myself.

**Comment by mperktold** [comment]
Nevermind, multiple clients are supported, you just need to provide different hostnames and different certificates for each of them.

**Comment by aliakbar-bsr** [comment]
Do we have to make multiple certificates in server side ?? And if yes how to make one that differs from the one that provided via command line in description ?

**Comment by mperktold** [comment]
@aliakbar-bsr no, you don't need multiple certificates on the server side. Every client needs to have it's own client certificate, that's all.

**Comment by aliakbar-bsr** [comment]
That's nice I'm going to try it soon 
Thanks for your awesome works on this projects
